Source/WebCore/accessibility/AXObjectCache.cpp

```diff
@@ -1304,6 +1304,14 @@ void AXObjectCache::handleChildrenChanged(AccessibilityObject& object)
     else if (auto* axRow = dynamicDowncast<AccessibilityTableRow>(object)) {
         if (auto* parentTable = axRow->parentTable())
             deferRecomputeTableCellSlots(*parentTable);
+    } else if (auto* scrollView = dynamicDowncast<AccessibilityScrollView>(object)) {
+        // When the children of an iframe change, e.g., because it's visibility changes,
+        // then we need to dirty the web area's subtree since the scroll area doesn't
+        // have a node nor renderer, thus, failing the check below and returning early.
+        if (RefPtr webArea = scrollView->webAreaObject(); webArea != rootWebArea()) {
```

Comment:
This is missing a nullptr check. It's one downside of these `;` checks in if-statements — you don't get the implicit null check. `webArea && webArea != rootWebArea()`.

Should the comment explain why we only do this when `webArea != rootWebArea()`?

Reply:
ah yes I can add that explanation!